src/modules/time.rs

```diff
@@ -1,39 +1,33 @@
-use ansi_term::Color;
 use chrono::{DateTime, Local};
 
 use super::{Context, Module};
 
+use crate::config::RootModuleConfig;
+use crate::configs::time::TimeConfig;
+
 /// Outputs the current time
 pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
-    let mut module = context.new_module("time");
-
-    if module.config_value_bool("disabled").unwrap_or(true) {
-        return None;
-    }
-
-    let module_style = module
-        .config_value_style("style")
-        .unwrap_or_else(|| Color::Yellow.bold());
-    module.set_style(module_style);
+    const TIME_PREFIX: &str = "at ";
 
-    // Load module settings
-    let is_12hr = module.config_value_bool("12hr").unwrap_or(false);
+    let mut module = context.new_module("time");
+    let config: TimeConfig = TimeConfig::try_load(module.config);
 
-    let default_format = if is_12hr { "%r" } else { "%T" };
-    let time_format = module
-        .config_value_str("format")
-        .unwrap_or(default_format)
-        .to_owned();
+    let default_format = if config.use_12hr { "%r" } else { "%T" };
+    let time_format = config.format.unwrap_or(default_format);
```

Comment:
Instead of using `format: Option<&str>` in the struct, you can use `format: &str` and provide a default format in `fn new()`, so that we can store all default value in the same place. What do you think?

Reply:
OK, my bad. It seems that default format depends on `use_12hr`.